pkg/reconciler/common/releases.go

```diff
@@ -221,6 +221,33 @@ func createManifestsPath(instance v1alpha1.KComponent) []string {
 	return nil
 }
 
+// componentIngressPath composes the ingress path for serving based on version
+func componentIngressPath(version string, instance v1alpha1.KComponent) string {
+	koDataDir := os.Getenv(KoEnvKey)
+	switch instance.(type) {
+	case *v1alpha1.KnativeServing:
```

Comment:
It's surprising to see any mention of "ingress" in the common package since it's strictly related to `KnativeServing`. I wouldn't expect any function operating on a `KComponent` to know anything about an ingress.

Reply:
I moved the functions into ingress package. They are called by the knativeserving reconciler.